pkgs/by-name/xa/xar/package.nix

```diff
@@ -0,0 +1,175 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  applyPatches,
+
+  autoreconfHook,
+
+  # Required dependencies.
+  zlib,
+  libxml2,
+
+  # Optional dependencies.
+  withExt2 ? lib.meta.availableOn stdenv.hostPlatform e2fsprogs,
+  e2fsprogs,
+  withAcl ? lib.meta.availableOn stdenv.hostPlatform acl,
+  acl,
+  withBzip2 ? true,
+  bzip2,
+  withLzma ? true,
+  xz, # lzma
```

Comment:
`availableOn stdenv.hostPlatform` should be true on Darwin then, shouldn’t it?

Reply:
Yes. My point is that both packages are `lib.platforms.unix`, so you might as well write `true` instead (and then perhaps not write the flag at all).

Reply:
`acl` package is not available on macOS, but `e2fsprogs` indeed is.

> Is this for bootstrap purposes?

I wasn’t really sure what set of packages was available for darwin stdenv bootstrap so I added a feature flag for each optional dependency. Turns out it builds fine, so unless someone is working on reducing the bootstrap chain for Darwin, we can drop these flags.

>unless you have ambitions of getting this to run on Windows

I actually tried building for Windows, but the code at least needs `uid_t` and that’s a sign of a potential rabbit hole so I didn’t investigate further.

Reply:
Yeah, I started looking into getting `acl` to build but it wasn’t clear if we need it (does Darwin just have those headers natively?).

If there’s no bootstrap issues I’d personally prefer dropping these flags as most likely being gratuitous API surface. Reducing the bootstrap chain would be noble, but is probably a different project altogether.

Reply:
I also tried to get `acl` to build a while back. The problem is the APIs for xattrs are different between Darwin and Linux. `attr` fails to build because it doesn’t support the Darwin ones.

For example, `listxattr` takes an options argument on Darwin but not Linux:
* https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/listxattr.2.html
* https://www.man7.org/linux/man-pages/man2/listxattr.2.html

Reply:
> does Darwin just have those headers natively?
I think it has native ACL implementation in Libc.
https://github.com/apple-oss-distributions/Libc/blob/Libc-1592.100.35/include/sys/acl.h